### PR TITLE
[discussion] Flatten the preparation of the provided ViewModels.

### PR DESCRIPTION
**Pull request by sys1yagi**
## Issue

none

## Overview (Required)

- Remove unnecessary provider nesting.  e748463
- Replaced ProvideViewModel Composable with a function that creates a ProvidedValue. 
131de2f

## Links

none

## Screenshot

none

### CODE REVIEW

File: uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/FeedViewModel.kt [hunk 1]
```diff
@@ -38,10 +38,8 @@ private val LocalFeedViewModel = compositionLocalOf<FeedViewModel> {
     error("not LocalFeedViewModel provided")
 }
 
-@Composable
-fun ProvideFeedViewModel(viewModel: FeedViewModel, block: @Composable () -> Unit) {
-    CompositionLocalProvider(LocalFeedViewModel provides viewModel, content = block)
-}
+fun feedViewModelProviderValue(viewModel: FeedViewModel) =
+    LocalFeedViewModel provides viewModel
```
Comment (sys1yagi) [inline uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/FeedViewModel.kt:42]: Replaced ProvideViewModel Composable with a function that creates a ProvidedValue.

File: uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/DroidKaigiApp.kt [hunk 2]
```diff
@@ -17,38 +17,36 @@ import io.github.droidkaigi.feeder.core.use
 
 @Composable
 fun DroidKaigiApp(firstSplashScreenState: SplashState = SplashState.Shown) {
-    ProvideAppViewModel(viewModel = appViewModel()) {
```
Comment (sys1yagi) [inline uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/DroidKaigiApp.kt:20]: Remove unnecessary provider nesting.

File: uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/viewmodel/ProvideRealViewModels.kt [hunk 3]
```diff
@@ -1,28 +1,25 @@
...
-                    ProvideContributorViewModel(viewModel = fakeContributorViewModel()) {
-                        ProvideFmPlayerViewModel(viewModel<RealFmPlayerViewModel>()) {
-                            content()
-                        }
-                    }
-                }
-            }
-        }
-    }
+    CompositionLocalProvider(
```
Comment (sys1yagi) [inline uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/viewmodel/ProvideRealViewModels.kt:16]: flatten!

**Comment by takahirom** [comment]
👀

File: uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/viewmodel/ProvideRealViewModels.kt (hunk 3 above)
Comment (takahirom) [reply uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/viewmodel/ProvideRealViewModels.kt:16]: 🆒

File: uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/DroidKaigiApp.kt (hunk 2 above)
Comment (takahirom) [reply uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/DroidKaigiApp.kt:20]: I didn't notice it! Do you have eagle eyes?  🦅  👍

**Comment by takahirom** [comment]
I think it's arguable how to provide ProvidedValue,
I think it's definitely better than the current situation. 👍

**Review by takahirom — APPROVED** [review]
LGTM 👍

**Pull request merged**